src/Universum/String/Conversion.hs

```diff
@@ -73,12 +74,21 @@ class ConvertUtf8 a b where
     -- патак
     decodeUtf8 :: b -> a
 
+#if MIN_VERSION_text(1,2,3)
+    -- | Decode as utf8 string but returning execption if byte sequence is malformed.
+    --
+    -- >>> decodeUtf8 @Text @ByteString "\208\208\176\209\130\208\176\208\186"
+    -- "\65533\1072\1090\1072\1082"
+    -- >>> decodeUtf8Strict @Text @ByteString "\208\208\176\209\130\208\176\208\186"
+    -- Left Cannot decode byte '\xd0': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream
+#else
     -- | Decode as utf8 string but returning execption if byte sequence is malformed.
     --
     -- >>> decodeUtf8 @Text @ByteString "\208\208\176\209\130\208\176\208\186"
     -- "\65533\65533\1090\1072\1082"
```

Comment:
Probably it's better to reduce some duplication and to put `if` only on this line. I guess it won't spoil readability, what do you think?

Reply:
@vrom911 You're right, this would be better. But I didn't manage to make it work... For some unknown reason if I put `#if` only for this single line tests doesn't work as expected. Probably `-XCPP` doesn't work well with comments 🙁 I didn't spend much time investigating this. Probably, if we use better CPP-preprocessor (like `cpphs`) we can avoid this problem. But I'm not sure I want to add `cpphs` to dependencies right now...

Reply:
I can try to do that

Reply:
@vrom911 I tried. If I put only those two lines under `#if` then doctest just ignores both test. I think it doesn't like empty lines in the middle of documentation... I guessed that the reason why it worked for `o-clock` is in the fact that `o-clock` uses block comments `{- | -}`. If I replace with block comments then everything works!

Reply:
Oh! I see now..